java-frontend/src/main/java/org/sonar/java/AbstractJavaClasspath.java

```diff
@@ -102,6 +103,10 @@ protected boolean hasJavaSources() {
     return fs.hasFiles(fs.predicates().and(fs.predicates().hasLanguage("java"), fs.predicates().hasType(fileType)));
   }
 
+  protected boolean hasMoreThanOneJavaFile() {
+    return Iterables.size(fs.files(fs.predicates().and(fs.predicates().hasLanguage("java"), fs.predicates().hasType(fileType)))) > 1;
```

Comment:
Problem is that you can perfectly analyze only sources (while analyzing only test is possible, it is less likely).